devito/types/utils.py

```diff
@@ -24,6 +24,8 @@ class Stagger(Tag):
 
 class DimensionTuple(EnrichedTuple):
 
+    __rkwargs__ = EnrichedTuple.__rkwargs__ + ('left', 'right')
```

Comment:
Can add a comment why this needed?

Reply:
if really needed, this would be a hack, which we cannot merge